src/porepy/models/solution_strategy.py

```diff
@@ -508,26 +516,78 @@ def check_convergence(
             # At least for the default direct solver, scipy.sparse.linalg.spsolve, no
             # error (but a warning) is raised for singular matrices, but a nan solution
             # is returned. We check for this.
-            diverged = bool(np.any(np.isnan(solution)))
+            diverged = bool(np.any(np.isnan(solution_increment)))
             converged: bool = not diverged
-            error: float = np.nan if diverged else 0.0
-            return error, converged, diverged
+            residual_error: float = np.nan if diverged else 0.0
+            increment_error: float = np.nan if diverged else 0.0
```

Comment:
It may be worth reflecting on the choice of naming here: Do we want to call the norm of the residual and increment `residual_error` and `increment_error` as right now? Or do we e.g. rather choose `norm_residual` and `norm_increment`. I actually rather prefer the second choice. Currently this PR rather consistently uses the former.

Reply:
I vote for second choice `norm_residual` and `norm_increment` or the inflections `residual_norm` and `increment_norm`.

Reply:
@mariusnevland will set up an overview which will summarize all chosen namings. After that an overall discussion will be initiated.

Reply:
Here is an overview of used/introduced terminology concerned with measuring norms for different files (as it is not consistent atm):

**File: src/porepy/models/solution_strategy.py**
pp.SolverStatistics: Object monitoring the performance of the nonlinear solver in terms of iterations, norms of increments and residuals as computed throughout the simulation
solution_increment (np.ndarray): solution vector to Newton's method
residual (np.ndarray): full residual vector 
init_residual (np.ndarray): full residual vector at first iteration
nonlinear_residual_error (method): Method to determine the residual_error (See below)
nonlinear_increment_error (method): Method to determine the increment_error (See below)
residual_error (float): norm of the residual
increment_error (float): norm of the increment
In the DOC it is also written 'update' or 'update increment' in some places. 

**File: src/porepy/numerics/linear_solvers.py**
error_res (float): norm of the residual
error_inc (float): norm of the increment

**File: src/porepy/numerics/nonlinear/nonlinear_solvers.py**
error_res (float): norm of the residual
error_inc (float): norm of the increment
sol (np.ndarray): increment vector
res (np.ndarray): residual vector
DOC: Use of 'Newton update increment'

**File: src/porepy/viz/solver_statistics.py**
increment_error (float): norm of increment vector
residual_error (Float): norm or residual vector
increment_errors (list): list of norms of increments vectors
residual_errors (list): list of norms of residual vectors

--------

To turn around the overview, here is a list of relevant quantities that ask for consistent naming:

residual vector
increment vector
norm of residual vector
norm of increment vector
method to determine norm of residual vector
method to determine norm of increment vector

I do not have strong opinions, but here are some suggestions, respectively:

residual
increment
residual_norm
increment_norm
determine_residual_norm()
determine_increment_norm()

Changes in solverstatistics would follow accordingly.

Reply:
I like your suggestion. Two questions:
1. Does it make sense to specify that this is the "nonlinear" increment?
2. Is "compute" instead of "determine" more consistent with rest of code?

Reply:
> I like your suggestion. Two questions:
> 
> 1. Does it make sense to specify that this is the "nonlinear" increment?

To me, yes. There will also be a time increment.

> 2. Is "compute" instead of "determine" more consistent with rest of code?

'Compute' fits better with how I picture the process (determine sounds more cumbersome and possibly less accurate), but I do not have very strong feelings here.

Reply:
Done as commented here now.